src/SimpleCacheTest.php

```diff
@@ -0,0 +1,589 @@
+            $this->markTestSkipped($this->skippedTests[__FUNCTION__]);
+
+            return;
+        }
+
+        $this->assertNull($this->cache->get('key'));
+        $this->assertEquals('foo', $this->cache->get('key', 'foo'));
+
+        $this->cache->set('key', 'value');
+        $this->assertEquals('value', $this->cache->get('key', 'foo'));
+    }
+
+    public function testDelete()
+    {
+        if (isset($this->skippedTests[__FUNCTION__])) {
+            $this->markTestSkipped($this->skippedTests[__FUNCTION__]);
+
+            return;
+        }
+
+        $this->assertTrue($this->cache->delete('key'), 'Deleting a value that does not exist should return true');
+        $this->cache->set('key', 'value');
+        $this->assertTrue($this->cache->delete('key'), 'Delete must return true on success');
+        $this->assertNull($this->cache->get('key'), 'Values must be deleted on delete()');
+    }
+
+    public function testClear()
+    {
+        if (isset($this->skippedTests[__FUNCTION__])) {
+            $this->markTestSkipped($this->skippedTests[__FUNCTION__]);
+
+            return;
+        }
+
+        $this->assertTrue($this->cache->clear(), 'Clearing an empty cache should return true');
+        $this->cache->set('key', 'value');
+        $this->assertTrue($this->cache->clear(), 'Delete must return true on success');
+        $this->assertNull($this->cache->get('key'), 'Values must be deleted on clear()');
+    }
+
+    public function testSetMultiple()
+    {
+        if (isset($this->skippedTests[__FUNCTION__])) {
+            $this->markTestSkipped($this->skippedTests[__FUNCTION__]);
+
+            return;
+        }
+
+        $result = $this->cache->setMultiple(['key0' => 'value0', 'key1' => 'value1']);
+        $this->assertTrue($result, 'setMultiple() must return true if success');
+        $this->assertEquals('value0', $this->cache->get('key0'));
+        $this->assertEquals('value1', $this->cache->get('key1'));
+
+        $this->cache->setMultiple(['key2' => 'value2', 'key3' => 'value3'], 1);
+        $this->assertEquals('value2', $this->cache->get('key2'));
+        $this->assertEquals('value3', $this->cache->get('key3'));
+        sleep(2);
+        $this->assertNull($this->cache->get('key2'), 'Value must expire after ttl.');
+        $this->assertNull($this->cache->get('key3'), 'Value must expire after ttl.');
+
+        $this->cache->setMultiple(['key4' => 'value4'], new \DateInterval('PT1S'));
+        $this->assertEquals('value4', $this->cache->get('key4'));
+        sleep(2);
+        $this->assertNull($this->cache->get('key4'), 'Value must expire after ttl.');
+    }
+
+    public function testSetMultipleWithGenerator()
+    {
+        if (isset($this->skippedTests[__FUNCTION__])) {
+            $this->markTestSkipped($this->skippedTests[__FUNCTION__]);
+
+            return;
+        }
+
+        $gen = function () {
+            yield 'key0' => 'value0';
+            yield 'key1' => 'value1';
+        };
```

Comment:
There is nothing wrong with a normal array. We test for that as well. This test is to make sure that generators are working.